unit-tests/py/rspy/devices.py

```diff
@@ -132,13 +133,70 @@ def query( monitor_changes = True ):
             else:
                 sn = dev.get_info( rs.camera_info.serial_number )
             device = Device( sn, dev )
+            if device.port is not None:
+                known_ports.append(device.port)
             _device_by_sn[sn] = device
             log.d( '... port {}:'.format( device.port is None and '?' or device.port ), dev )
     finally:
         log.debug_unindent()
     #
     if monitor_changes:
         _context.set_devices_changed_callback( _device_change_callback )
+    if acroname:
+        devices_with_unknown_ports = [device for device in _device_by_sn.values() if device.port is None]
+        if devices_with_unknown_ports:
+            log.d( 'trying do discover unknown ports' )
+            ports = acroname.ports()
+            unknown_ports = [port for port in ports if port not in known_ports]
+            log.d( "printing active ports" )
+            log.debug_indent()
+            log.d( "all active ports:", ports )
+            log.d( "ports with known devices:", known_ports )
+            log.d( "unknown ports:", unknown_ports )
+            log.debug_unindent()
```

Comment:
Should be at the end of the function in a `finally:`

Reply:
but then all the following debug messages will also be indented, is that what we want?

Reply:
all the `port <#> has device <sn>` messages

Reply:
Yep